lib/engine/step/g_18_co/moving_bid_auction.rb

```diff
@@ -0,0 +1,215 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+require_relative '../auctioner'
+
+module Engine
+  module Step
+    module G18CO
+      class MovingBidAuction < Base
+        include Auctioner
+        ACTIONS = %w[bid move_bid pass].freeze
+
+        attr_reader :companies
+
+        def description
+          'Moving Bid Auction for Companies'
+        end
+
+        def available
+          @companies
+        end
+
+        def process_pass(action)
+          entity = action.entity
+          @log << "#{entity.name} passes bidding"
+          entity.pass!
+          all_passed! if entities.all?(&:passed?)
+          @round.next_entity_index!
+        end
+
+        def process_bid(action)
+          add_bid(action)
+          action.entity.unpass!
+          @round.next_entity_index!
+        end
+
+        def process_move_bid(action)
+          move_bid(action)
+          action.entity.unpass!
+          @round.next_entity_index!
+        end
+
+        def actions(_entity)
+          return [] if entities.all?(&:passed?)
+
+          ACTIONS
+        end
+
+        def setup
+          setup_auction
+          @companies = @game.companies.sort_by(&:value)
+        end
+
+        def round_state
+          {
+            companies_pending_par: [],
+          }
+        end
+
+        def auctioning
+          nil
+        end
+
+        # min bid is face value or $5 higher than previous bid
+        def min_bid(company)
+          return unless company
+
+          high_bid = highest_bid(company)
+          (high_bid ? high_bid.price + min_increment : company.min_bid)
+        end
+
+        # can never purchase directly
+        def may_purchase?(_company)
+          false
+        end
+
+        def committed_cash(player, _show_hidden = false)
+          bids = bids_for_player(player)
+          return 0 if bids.empty?
+
+          bids.sum(&:price)
+        end
+
+        def current_bid_amount(player, company)
+          @bids[company]&.select { |b| b.entity == player }&.sort(&:price)&.last&.price || 0
```

Comment:
maybe do a safe return first 

return unless (bids = @bids[company])

bids.select...min(&:price)  || 0

Reply:
I think we want max, not min ... otherwise yeah that's cleaner.